components/post_view/message_attachments/action_button/action_button.test.jsx

```diff
@@ -0,0 +1,75 @@
+import {changeOpacity} from 'mattermost-redux/utils/theme_utils';
+
+import {Constants, STATUS_COLORS} from 'utils/constants';
+import ActionButton from 'components/post_view/message_attachments/action_button/action_button';
+
+describe('components/post_view/message_attachments/action_button.jsx', () => {
+    const baseProps = {
+        action: {id: 'action_id_1', name: 'action_name_1', cookie: 'cookie-contents'},
+        handleAction: jest.fn(),
+        theme: Constants.THEMES.default,
+    };
+
+    test('should match snapshot', () => {
+        const wrapper = shallow(<ActionButton {...baseProps}/>);
+        expect(wrapper).toMatchSnapshot();
+    });
+
+    test('should call handleAction on click', () => {
+        const wrapper = shallow(<ActionButton {...baseProps}/>);
+
+        wrapper.find('button').simulate('click');
+
+        expect(baseProps.handleAction).toHaveBeenCalledTimes(1);
+    });
+
+    test('should have correct styles when provided color from theme', () => {
+        const props = {
+            ...baseProps,
+            action: {...baseProps.action, style: 'onlineIndicator'},
+        };
+
+        const wrapper = shallow(<ActionButton {...props}/>);
+        const buttonStyles = wrapper.find('button').prop('style');
+
+        expect(buttonStyles).toHaveProperty('backgroundColor', '#ffffff');
+        expect(buttonStyles).toHaveProperty('borderColor', changeOpacity(Constants.THEMES.default.onlineIndicator, 0.25));
+        expect(buttonStyles).toHaveProperty('borderWidth', 2);
+        expect(buttonStyles).toHaveProperty('color', Constants.THEMES.default.onlineIndicator);
+    });
+
+    test('should have correct styles when provided status color', () => {
+        const props = {
+            ...baseProps,
+            action: {...baseProps.action, style: 'success'},
+        };
+
+        const wrapper = shallow(<ActionButton {...props}/>);
+        const buttonStyles = wrapper.find('button').prop('style');
+
+        expect(buttonStyles).toHaveProperty('backgroundColor', '#ffffff');
+        expect(buttonStyles).toHaveProperty('borderColor', changeOpacity(STATUS_COLORS[props.action.style], 0.25));
+        expect(buttonStyles).toHaveProperty('borderWidth', 2);
+        expect(buttonStyles).toHaveProperty('color', STATUS_COLORS[props.action.style]);
+    });
+
+    test('should have correct styles when provided hex color', () => {
+        const props = {
+            ...baseProps,
+            action: {...baseProps.action, style: '#28a745'},
+        };
+
+        const wrapper = shallow(<ActionButton {...props}/>);
+        const buttonStyles = wrapper.find('button').prop('style');
+
+        expect(buttonStyles).toHaveProperty('backgroundColor', '#ffffff');
+        expect(buttonStyles).toHaveProperty('borderColor', changeOpacity(props.action.style, 0.25));
+        expect(buttonStyles).toHaveProperty('borderWidth', 2);
+        expect(buttonStyles).toHaveProperty('color', props.action.style);
+    });
```

Comment:
Following one of my previous comments, should test what happens when something is wrong (specially, action.style) not being a proper color.

Reply:
I added two new test cases: when the action.style has a wrong color and when it's undefined :+1: